server/server.go

```diff
@@ -145,17 +146,30 @@ func (s *Server) handle(conn net.Conn) {
 }
 
 func (s *Server) doHandle(conn net.Conn, isWrapConn bool, wrapConn listeners.WrapConn) {
+	log.Debug(conn)
+	log.Debug(conn.RemoteAddr())
+	clientIP := ""
+	remoteAddr := conn.RemoteAddr()
+	if remoteAddr != nil {
+		clientIP, _, _ = net.SplitHostPort(remoteAddr.String())
+	}
+	op := ops.Begin("http_proxy_handle").Set("client_ip", clientIP)
```

Comment:
Do we maybe only want to trigger ops in cases where we definitely get a client IP?

Reply:
The only case where we won't get a client IP is during unit testing where we sometimes use mock connections. In production, there should always be a client IP.